raiden/messages/transfers.py

```diff
@@ -406,19 +412,95 @@ def from_event(cls, event: Any) -> Any:  # noqa: F811
         )
 
     def _packed_data(self) -> bytes:
+        return LockedTransferBase._pack_locked_transfer_data(
+            self.cmdid,  # pylint: disable=no-member
+            self.message_identifier,
+            self.payment_identifier,
+            self.token,
+            self.recipient,
+            self.target,
+            self.initiator,
+            self.lock,
+        )
+
+    @classmethod
+    def _pack_locked_transfer_data(
+        cls,
+        cmdid: CmdId,
+        message_identifier: MessageID,
+        payment_identifier: PaymentID,
+        token: TokenAddress,
+        recipient: Address,
+        target: TargetAddress,
+        initiator: InitiatorAddress,
+        lock: Lock,
+    ) -> bytes:
         return (
-            bytes([self.cmdid.value])  # pylint: disable=no-member
-            + self.message_identifier.to_bytes(8, byteorder="big")
-            + self.payment_identifier.to_bytes(8, byteorder="big")
-            + self.lock.expiration.to_bytes(32, byteorder="big")
-            + self.token
-            + self.recipient
-            + self.target
-            + self.initiator
-            + self.lock.secrethash
-            + self.lock.amount.to_bytes(32, byteorder="big")
+            bytes([cmdid.value])
+            + message_identifier.to_bytes(8, byteorder="big")
+            + payment_identifier.to_bytes(8, byteorder="big")
+            + lock.expiration.to_bytes(32, byteorder="big")
+            + token
+            + recipient
+            + target
+            + initiator
+            + lock.secrethash
+            + lock.amount.to_bytes(32, byteorder="big")
+        )
+
+    @post_load(pass_original=True)
+    def _check_metadata_hash(  # pylint: disable=no-self-use,unused-argument
+        self, data: Dict[str, Any], original_data: Dict[str, Any], many: bool, **kwargs: Any
+    ) -> Dict[str, Any]:
+
+        metadata = data["metadata"]
+        expected_signer = original_data.get("peer_address", None)
+        if expected_signer is None:
+            return data
+
+        balance_hash = hash_balance_data(
+            data["transferred_amount"], data["locked_amount"], data["locksroot"]
         )
 
+        packed_data = LockedTransferBase._pack_locked_transfer_data(
```

Comment:
Unfortunately it is the latter... self is an object of schema, which has the decorator functions copied over but everything else not. So `self` is not an object of LockedTransfer (which totally makes sense because it is not deserialized in a `post_load()` hook).

Reply:
In order to reuse packing the data I needed to make it either a classmethod or a utility method outside of the class bnut not an object method.